modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -49,6 +59,67 @@
 				title: editor.lang.link.title,
 			});
 
+			editor.ui.addBalloonToolbarButton('Twitter', {
+				_MAX_TWEET_LENGTH: 280,
+				_getHref(url, via) {
+					const selectedText = editor
+						.getSelection()
+						.getSelectedText()
+						.substring(0, this._MAX_TWEET_LENGTH);
+
+					// TODO - These parameters should be revisited when to use `url` and `via`.
+					// const url = this.props.url;
+					// const via = this.props.via;
+
+					let twitterHref =
+						'https://twitter.com/intent/tweet?text=' + selectedText;
+
+					if (url) {
+						twitterHref += '&url=' + url;
+					}
+
+					if (via) {
+						twitterHref += '&via=' + via;
+					}
+
+					return twitterHref;
+				},
+				_isActive() {
+					const link = linkUtils.getFromSelection();
+
+					return (
+						link &&
+						link
+							.getAttribute('href')
+							.indexOf('twitter.com/intent/tweet') !== -1
+					);
+				},
+				click() {
+					if (this._isActive()) {
+						linkUtils.remove(linkUtils.getFromSelection());
+					}
+					else {
+						const linkInputValue = linkUtils.create(
+							this._getHref(),
+							{
+								class: 'ae-twitter-link',
```

Comment:
I need to remove this config here because I'm passing target using the config obj when emitting showToolbar event